sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py

```diff
@@ -252,6 +242,152 @@ def begin_restore(  # pylint: disable=docstring-missing-param,docstring-keyword-
             **kwargs,
         )
 
+    @overload
+    def begin_pre_backup(
+        self,
+        blob_storage_url: str,
+        *,
+        use_managed_identity: Literal[True],
+        continuation_token: Optional[str] = None,
+        **kwargs: Any,
+    ) -> LROPoller[KeyVaultBackupOperation]:
+        ...
+
+    @overload
+    def begin_pre_backup(
+        self,
+        blob_storage_url: str,
+        *,
+        sas_token: str,
+        continuation_token: Optional[str] = None,
+        **kwargs: Any,
+    ) -> LROPoller[KeyVaultBackupOperation]:
+        ...
+
+    @distributed_trace
+    def begin_pre_backup(  # pylint: disable=docstring-keyword-should-match-keyword-only
+        self, blob_storage_url: str, **kwargs: Any
+    ) -> LROPoller[KeyVaultBackupOperation]:
+        """Initiates a pre-backup check of whether a full Key Vault backup can be performed.
+
+        A :class:`KeyVaultBackupOperation` instance will be returned by the poller's `result()` method. If the
+        pre-backup check is successful, the object will have a string `folder_url` attribute, pointing to the blob
+        storage container where the backup will be stored. If the check fails, the object will have a string `error`
+        attribute.
+
+        :param str blob_storage_url: URL of the blob storage container in which the backup will be stored, for example
+            https://<account>.blob.core.windows.net/backup.
+
+        :keyword str sas_token: Optional Shared Access Signature (SAS) token to authorize access to the blob. Required
+            unless `use_managed_identity` is set to True.
+        :keyword use_managed_identity: Indicates which authentication method should be used. If set to True, Managed HSM
+            will use the configured user-assigned managed identity to authenticate with Azure Storage. Otherwise, a SAS
+            token has to be specified.
+        :paramtype use_managed_identity: bool
+        :keyword str continuation_token: A continuation token to restart polling from a saved state.
+
+        :returns: An :class:`~azure.core.polling.LROPoller` instance. Call `result()` on this object to wait for the
+            operation to complete and get a :class:`KeyVaultBackupOperation`. If the pre-backup check is successful, the
+            object will have a string `folder_url` attribute, pointing to the blob storage container where the backup
+            will be stored. If the check fails, the object will have a string `error` attribute.
+        :rtype: ~azure.core.polling.LROPoller[~azure.keyvault.administration.KeyVaultBackupOperation]
+        """
+        polling_interval: int = kwargs.pop("_polling_interval", 5)
```

Comment:
why the `_` for polling_interval. Is it supposed to be hidden ?

Reply:
Yes, this is the pattern throughout KV; for example, in [KeyVaultBackupClient.begin_backup](https://github.com/Azure/azure-sdk-for-python/blob/31915bca912b05d9d9f214986cae333a56c95e2b/sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_backup_client.py#L95) and [KeyClient.begin_delete_key](https://github.com/Azure/azure-sdk-for-python/blob/3d9629b153812c3afebaf440bfea1d1b3f212815/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py#L428). I'm not actually clear on why that was originally settled on as the pattern though, so I might follow up on that -- for now though, just following convention 🙂